### DISCUSSION TITLE
Macros takes a lot of CPU and build time

### DISCUSSION THREAD

**Discussion by arnauddorgans** [pointfreeco/swift-composable-architecture#2987] [Q&A]
Is there a way to reduce the workload of macros? 
I have been using TCA since a few months now and the build times have become longer and longer, I almost never have auto-completion while typing my code now 😢 

<img width="368" alt="Screenshot 2024-04-13 at 20 13 26" src="https://github.com/pointfreeco/swift-composable-architecture/assets/10015557/6444fd8d-0399-49fc-9475-a137bb92bde8">

**Comment by mbrandonw** [comment]
Hi @arnauddorgans, unfortunately this is really a problem for Apple to solve in Swift. There have been a few posts about it on the forums ([here](https://forums.swift.org/t/macro-adoption-concerns-around-swiftsyntax/66588) and [here](https://forums.swift.org/t/compilation-extremely-slow-since-macros-adoption/67921)), and so it would be good for you to chime in on one of those threads to make it clear how problematic this has become.

Some people have said Tuist is capable of easing the build time penalty of macros, but we aren't personally familiar with it. And also it won't help with the CPU performance that your screenshot shows.

**Reply by arnauddorgans** [reply]
I can confirm that Tuist reduced the build time by a lot 👍
